### PR TITLE
feat(universities): add university clients

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* Add a new super client for all the involved universities
* New generic university careers domain

## Why

* 4 new universities all using the same service as HÍ. Wasteful to make a new client for each since they're identical, so we just make a big client instead. 
* Also new api domain that's more generic.
* new function for getting contentful organization by reference id

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by busla — APPROVED** [review]
infra LGTM

File: libs/cms/src/lib/cms.contentful.service.ts [hunk 1]
```diff
@@ -345,6 +345,25 @@ export class CmsContentfulService {
...
   }
 
+  async getOrganizationByReferenceId(
+    referenceId: string,
+    lang: string,
+  ): Promise<Organization> {
+    const params = {
+      ['content_type']: 'organization',
+      include: 10,
+      'fields.referenceIdentifier[match]': referenceId,
```
Comment (RunarVestmann) [inline libs/cms/src/lib/cms.contentful.service.ts:355]: Just wanted to let you know that the [match] will also match those that have the same prefix (not sure if that will ever be an issue though). I wonder if

`'fields.referenceIdentifier': referenceId`

makes more sense 🤔

**Review by RunarVestmann — APPROVED** [review]
I'll approve the 1 file that stefna owns but have a comment

File: libs/cms/src/lib/cms.contentful.service.ts (hunk 1 above)
Comment (thorkellmani) [reply libs/cms/src/lib/cms.contentful.service.ts:355]: Ah ok, I'll try it

Comment (thorkellmani) [reply libs/cms/src/lib/cms.contentful.service.ts:355]: Changed, resolved

**Review by obmagnusson — APPROVED** [review]
Codeowned file LGTM

**Review by Valur — APPROVED** [review]

**Review by saevarma — APPROVED** [review]

File: libs/api/domains/university-careers/src/lib/universityCareers.service.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+    @Inject(LOGGER_PROVIDER) private readonly logger: Logger,
+  ) {}
+
+  private getContentfulOrganizations = async (locale: Locale) => {
+    const universityIds = Object.values(UniversityId)
+    const referenceIds = universityIds.map(
+      (id) => UniversityContentfulReferenceIds[id],
+    )
+
+    //Get all organizations now -> fewer contentful calls
```
[the remark this answers is not in the document]
Comment (thorkellmani) [reply libs/api/domains/university-careers/src/lib/universityCareers.service.ts:46]: Ill check it out

**Pull request merged**